src/Mizar/SamplingWithFrameTrackInputWidget.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_SAMPLING_WITH_FRAME_TRACK_INPUT_WIDGET_H_
+#define MIZAR_SAMPLING_WITH_FRAME_TRACK_INPUT_WIDGET_H_
+
+#include <QWidget>
+#include <memory>
+#include <string_view>
+
+#include "MizarData/MizarPairedData.h"
+
+namespace Ui {
+class SamplingWithFrameTrackInputWidget;
+}
+
+namespace orbit_mizar {
+
+template <typename PairedData>
+class SamplingWithFrameTrackInputWidgetTmpl : public QWidget {
+ public:
+  explicit SamplingWithFrameTrackInputWidgetTmpl(QWidget* parent);
+  ~SamplingWithFrameTrackInputWidgetTmpl() override;
+
+  void Init(const PairedData* data, const QString& name);
+
+ private:
+  std::unique_ptr<Ui::SamplingWithFrameTrackInputWidget> ui_;
+};
```

Comment:
Classes that derive from `QObject` shouldn't/can't be templated because this is not supported by the MOC preprocessor.

Couldn't you just template `Init`?

Reply:
If not I would suggest to not give this widget direct access to `PairedData` but rather define an interface - similar to what Qt does with `QAbstractItemModel`.

Reply:
(Same goes for the other class)

Reply:
> Couldn't you just template Init?

I can try. But this way I will not be able to store a pointer to PairedData instance in the widget. 

>(Same goes for the other class)

Sorry, which class? I think, it's the only template class I add in the PR.

Reply:
`Function template partial specialization is not allowed`
... I guess, I can't.

Reply:
Done. The method is templated, the class isn't.